crypto/fuzzer/fuzzer.c

```diff
@@ -358,6 +372,339 @@ int fuzz_nem_validate_address() {
   return 0;
 }
 
+extern bool nem_get_address(const ed25519_public_key public_key,
+                            uint8_t version, char *address);
+
+int fuzz_nem_get_address(void) {
+  unsigned char ed25519_public_key[32] = {0};
+  uint32_t network = 0;
+
+  if (fuzzer_length != (sizeof(ed25519_public_key) + sizeof(network))) {
+    return 0;
+  }
+
+  char address[NEM_ADDRESS_SIZE + 1] = {0};
+
+  memcpy(ed25519_public_key, fuzzer_input(32), 32);
+  memcpy(&network, fuzzer_input(4), 4);
+
+  nem_get_address(ed25519_public_key, network, address);
+
+  // TODO check return address for memory info leakage?
+  return 0;
+}
+
+extern void xmr_get_subaddress_secret_key(bignum256modm r, uint32_t major,
+                                          uint32_t minor,
+                                          const bignum256modm m);
+
+int fuzz_xmr_get_subaddress_secret_key(void) {
+  bignum256modm m = {0};
+  uint32_t major = 0;
+  uint32_t minor = 0;
+  if (fuzzer_length != (sizeof(bignum256modm) + 2 * sizeof(uint32_t))) {
+    return 0;
+  }
+
+  bignum256modm output = {0};
+
+  memcpy(m, fuzzer_input(sizeof(bignum256modm)), sizeof(bignum256modm));
+  memcpy(&major, fuzzer_input(sizeof(uint32_t)), sizeof(uint32_t));
+  memcpy(&minor, fuzzer_input(sizeof(uint32_t)), sizeof(uint32_t));
+
+  xmr_get_subaddress_secret_key(output, major, minor, m);
+
+  return 0;
+}
+
+extern void xmr_derive_private_key(bignum256modm s, const ge25519 *deriv,
+                                   uint32_t idx, const bignum256modm base);
```

Comment:
Why are there so many external functions declared in the code? Usually external functions are declared by including the appropriate header file. Declaring them this way would make sense only if they were not in a header file, i.e. they are considered internal to the c-file, but then they would probably be static anyway and even if they weren't static, it would be questionable if such functions should be fuzzed.

Reply:
This is a valid question. My original rationale here was that it helps to have an explicit reference to which function is fuzzed including the function parameter definition. Using them as part of the `extern` definition made the reference functional, vs. just having a comment.

However, I agree that this notation is not very elegant as the number of target functions grows, and developers with a reasonably advanced IDE have automated access to this information anyway. So I've changed this to regular `#include` definitions already in the next iteration of this fuzzer collection.

@andrewkozlik : I'll update this PR. Thanks for the review so far.

Reply:
@andrewkozlik : I've changed the declaration style via ccac25e, so this PR is ready for review again.